contracts/external/actions/TreasuryAction.sol

```diff
@@ -422,15 +422,16 @@ contract TreasuryAction is StorageLayoutV2, ActionGuards, NotionalTreasury {
             // If this is zero then there is no outstanding lending.
             isExternalLendingUnhealthy = false;
         } else {
+            // offTargetPercentage = |currentExternalUnderlyingLend - targetAmount| * 100 / targetAmount
             uint256 offTargetPercentage = oracleData.currentExternalUnderlyingLend.toInt()
                 .sub(targetAmount.toInt()).abs()
                 .toUint()
                 .mul(uint256(Constants.PERCENTAGE_DECIMALS))
-                .div(targetAmount.add(oracleData.currentExternalUnderlyingLend));
+                .div(targetAmount + 1); // add 1 to prevent dividing by zero
```

Comment:
If targetAmount == 0 and currentExternalUnderlyingLend > 0 won't this always trigger isExternalLendingUnhealthy?

`isExternalLendingUnhealthy = (0 < 1 && (1 - 0 / 1) > 0`

Maybe we should add an absolute floor to the `|currentExternalUnderlyingLend - targetAmount` as well. In this code block it is underlying precision so we can set the floor to something like 1 basis point of 1 unit:

`minDiff = underlyingToken.decimals.mulInRatePrecision(Constants.BASIS_POINT)`

Reply:
What situation are you trying to protect against? If targetAmount == 0 why shouldn't we redeem if currentExternalUnderlyingLend != 0? As long as there isn't some situation where the amount is too small to redeem so there's just an infinite loop of the bot trying and failing to redeem and burning gas, what's the issue?